### PR TITLE
Return io.eof when buf size is larger than read size

### PR DESCRIPTION
**Pull request by yunxing**

### CODE REVIEW

**Comment by wangtuanjie** [comment]
issue  https://github.com/c-fs/cfs/issues/77

File: disk/disk.go [hunk 1]
```diff
@@ -58,6 +58,9 @@ func (d *Disk) ReadAt(name string, p []byte, off int64) (int, error) {
 		// 2. We read a partial block -- reach the end of the file
 		// 3. We can't copy into p anymore -- p is filled up
 		if err != nil || n < bsize-crc32Len || len(p) == 0 {
```
Comment (wangtuanjie) [inline disk/disk.go]: this, more clearly ?

``` go
if err != nil || len(p) == 0 {
    return read + copied, err
}
```

File: disk/disk_test.go [hunk 2]
```diff
@@ -453,3 +453,33 @@ func TestMkDirNonExistParent(t *testing.T) {
...
+		for i := int64(0); i < tt.writeLen; i++ {
+			p[i] = 'X'
+		}
+		d.WriteAt(tt.fileName, p, 0)
+		buf := make([]byte, tt.readLen)
+		read, err := d.ReadAt(tt.fileName, buf, 0)
+		if int64(read) != tt.writeLen {
+			t.Errorf("%d: read %d bytes != %d bytes written", i, read, tt.readLen)
+		}
+		if err != io.EOF {
```
Comment (wangtuanjie) [inline disk/disk_test.go:481]: CI failed because not import io package

File: disk/disk.go (hunk 1 above)
Comment (yunxing) [inline disk/disk.go]: Good point!

File: disk/disk_test.go (hunk 2 above)
Comment (yunxing) [inline disk/disk_test.go:481]: oops..

**Comment by wangtuanjie** [comment]
LGTM

**Pull request merged**